pkg/providers/aws/provider_redis.go

```diff
@@ -213,20 +214,23 @@ func (p *AWSRedisProvider) deleteRedisCluster(cacheSvc elasticacheiface.ElastiCa
 		}
 		return "redis cache successfully deleted", nil
 	}
+
+	// check and verify delete config
+	if err := p.buildRedisDeleteConfig(ctx, *r, redisCreateConfig, redisDeleteConfig); err != nil {
+		msg := "failed to verify aws rds instance configuration"
+		return v1alpha1.StatusMessage(msg), errorUtil.Wrap(err, msg)
+	}
 	// check if replication group exists and is available
 	if *foundCache.Status == "available" {
```

Comment:
I think if we inverse this we can flatten out the delete function. If the status is unavailable we should just return a phase message. Otherwise do everything else.